### PR TITLE
Update gvsbuild to upgrade cairo in Windows

### PR DESCRIPTION
**Pull request by danyeaw**
This PR fixes Windows build issues by:
- Upgrading gvsbuild to version 2022.6.0
- Update the Windows docs to use the official Python installer and install wheels from gvsbuild
- Fix an issue with `test_gaphorconvert` where it was using relative file paths, updated to use importlib

### PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I have read, and I am following the [Contributor guide](https://github.com/gaphor/gaphor/blob/main/CONTRIBUTING.md)
- [X] I have read, and I understand the [Code of Conduct](https://github.com/gaphor/gaphor/blob/main/CODE_OF_CONDUCT.md)

### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [ ] Feature
- [X] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

Issue Number: N/A

### Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

File: .github/workflows/full-build.yml [hunk 1]
```diff
@@ -264,12 +264,9 @@ jobs:
         run: python -m pip install gvsbuild==${{ env.gvsbuild_version }}
       - name: GTK binaries run gvsbuild
         if: steps.cache.outputs.cache-hit != 'true'
-        run: >
-          gvsbuild build --ninja-opts -j2 --enable-gi --py-wheel gobject-introspection gtk4 libadwaita pycairo pygobject
+        run: > # j2 option resolves out of memory issues while linking on GitHub Actions runners
```
Comment (amolenaar) [inline .github/workflows/full-build.yml:267]: I would put the comment on a new line, above the previous run statement.

**Review by amolenaar — APPROVED** [review]

**Comment by danyeaw** [comment]
I think I have resolved some of the issues by ensuring we are using the wheels built with gvsbuild. However, I can consistently reproduce an issue with the FTA Majority Vote gate. Either it is a problem with how we are drawing it, or possibly their is a bug in the latest version of Cairo:

```
Traceback (most recent call last):
  File "C:\Users\dan\Projects\gaphor\gaphor\diagram\painter.py", line 33, in paint_item
    item.draw(
  File "C:\Users\dan\Projects\gaphor\gaphor\diagram\presentation.py", line 152, in draw
    self._shape.draw(
  File "C:\Users\dan\Projects\gaphor\gaphor\diagram\shapes.py", line 307, in draw
    self.icon.draw(new_context, Rectangle(x, y, w, h))
  File "C:\Users\dan\Projects\gaphor\gaphor\diagram\shapes.py", line 200, in draw
    self._draw_border(self, new_context, bounding_box)
  File "C:\Users\dan\Projects\gaphor\gaphor\RAAML\fta\majorityvotegate.py", line 94, in draw_majority_vote_gate
    cr.select_font_face("Sans", cairo.FONT_SLANT_NORMAL, cairo.FONT_WEIGHT_NORMAL)
gaphor.RAAML.fta.majorityvotegate.cairo.MemoryError: out of memory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\dan\Projects\gaphor\.venv\Lib\site-packages\gaphas\decorators.py", line 86, in async_wrapper
    func(*args, **kwargs)
  File "C:\Users\dan\Projects\gaphor\.venv\Lib\site-packages\gaphas\view\gtkview.py", line 324, in update
    self.update_bounding_box(dirty_items)
  File "C:\Users\dan\Projects\gaphor\.venv\Lib\site-packages\gaphas\view\gtkview.py", line 361, in update_bounding_box
    painter.paint_item(item, cr)
  File "C:\Users\dan\Projects\gaphor\gaphor\diagram\painter.py", line 45, in paint_item
    cr.restore()
gaphor.diagram.painter.cairo.MemoryError: out of memory
gaphor.plugins.errorreports.errorreports ERROR Uncaught exception
Traceback (most recent call last):
  File "C:\Users\dan\Projects\gaphor\.venv\Lib\site-packages\gaphas\decorators.py", line 98, in async_wrapper
    func(*args, **kwargs)
  File "C:\Users\dan\Projects\gaphor\gaphor\diagram\tools\placement.py", line 117, in open_editor
    instant_editor(item, view, event_manager)
  File "C:\Users\dan\AppData\Local\Programs\Python\Python311\Lib\functools.py", line 909, in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dan\Projects\gaphor\gaphor\diagram\instanteditors.py", line 41, in named_item_editor
    box = view.get_item_bounding_box(item)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dan\Projects\gaphor\.venv\Lib\site-packages\gaphas\view\gtkview.py", line 289, in get_item_bounding_box
    *transform_rectangle(self._matrix, self._qtree.get_bounds(item))
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dan\Projects\gaphor\.venv\Lib\site-packages\gaphas\quadtree.py", line 200, in get_bounds
    return self._ids[item][0]
           ~~~~~~~~~^^^^^^
KeyError: <gaphor.RAAML.fta.majorityvotegate.MajorityVoteItem element 3eeecaac-7972-11ed-ba88-8caf892b5138>
```

File: .github/workflows/full-build.yml [hunk 2]
```diff
@@ -284,7 +287,7 @@ jobs:
     runs-on: windows-latest
     env:
       GAPHOR_USE_GTK: 4
-      INCLUDE: C:\gtk-build\gtk\x64\release\lib\glib-2.0\include;C:\gtk-build\gtk\x64\release\include;C:\gtk-build\gtk\x64\release\include\gobject-introspection-1.0;C:\gtk-build\gtk\x64\release\include\cairo;C:\gtk-build\gtk\x64\release\include\glib-2.0
+      INCLUDE: C:\gtk-build\gtk\x64\release\include;C:\gtk-build\gtk\x64\release\include\cairo;C:\gtk-build\gtk\x64\release\include\glib-2.0;C:\gtk-build\gtk\x64\release\include\gobject-introspection-1.0;C:\gtk-build\gtk\x64\release\lib\glib-2.0\include;
```
Comment (amolenaar) [inline .github/workflows/full-build.yml:290]: Would it make sense to get libs and includes via `pkg-config`?

File: docs/windows.md [hunk 3]
```diff
@@ -70,26 +70,45 @@ choco install visualstudio2022-workload-vctools
...
-Download and install the latest version of Python:
+In Windows, The full installer contains all the Python components and is the
+best option for developers using Python for any kind of project.
+
+For more information on how to use the official installer, please see the
+[full installer instructions](https://docs.python.org/3/using/windows.html#windows-full).
+The default installation options should be fine for use with Gaphor.
+
+1. Install the latest Python version using the
+[official installer](https://www.python.org/downloads/windows/).
```
Comment (amolenaar) [inline docs/windows.md:81]: Does this imply that installing Python via Choco or the Windows Store does not work?

File: docs/windows.md [hunk 4]
```diff
@@ -70,26 +70,45 @@ choco install visualstudio2022-workload-vctools
...
+
+#### Install Graphviz
+
+Graphviz is used by Gaphor for automatic diagram formatting.
 
 1. Install from Chocolately with administrator PowerShell:
 
    ```PowerShell
-   choco install python
+   choco install graphviz
```
Comment (amolenaar) [inline docs/windows.md:96]: Maybe we should build graphviz as part of gvsbuild? So it does not ship with it's own libraries?

For now it's fine, though.

**Review by amolenaar — APPROVED** [review]
Quite an achievement! Your perseverance is inspiring.

I added a few comments, but those are more questions from my side and do not need a change per se.

File: docs/windows.md (hunk 3 above)
Comment (danyeaw) [reply docs/windows.md:81]: It should work. I spent a lot of time messing with choco version and pyenv-win. I came to the conclusions that the Python official version is the best for Windows, especially if you need to have multiple versions installed. So I thought I would update our instructions to use this best practice.

File: docs/windows.md (hunk 4 above)
Comment (danyeaw) [reply docs/windows.md:96]: Yes, I think so as well. I'll start looking in to this :+1:

**Pull request merged**